L1Trigger/TrackFindingTMTT/src/Settings.cc

```diff
@@ -0,0 +1,508 @@
+        kf_phi0Range_(trackDigi_.getParameter<double>("KF_phi0Range")),
+        kf_z0Bits_(trackDigi_.getParameter<unsigned int>("KF_z0Bits")),
+        kf_z0Range_(trackDigi_.getParameter<double>("KF_z0Range")),
+        kf_tanlambdaBits_(trackDigi_.getParameter<unsigned int>("KF_tanlambdaBits")),
+        kf_tanlambdaRange_(trackDigi_.getParameter<double>("KF_tanlambdaRange")),
+        kf_chisquaredBits_(trackDigi_.getParameter<unsigned int>("KF_chisquaredBits")),
+        kf_chisquaredRange_(trackDigi_.getParameter<double>("KF_chisquaredRange")),
+        kf_chisquaredBinEdges_(trackDigi_.getParameter<vector<double>>("KF_chisquaredBinEdges")),
+        //
+        other_skipTrackDigi_(trackDigi_.getParameter<bool>("Other_skipTrackDigi")),
+
+        // Debug printout
+        debug_(iConfig.getParameter<unsigned int>("Debug")),
+        resPlotOpt_(iConfig.getParameter<bool>("ResPlotOpt")),
+        iPhiPlot_(iConfig.getParameter<unsigned int>("iPhiPlot")),
+        iEtaPlot_(iConfig.getParameter<unsigned int>("iEtaPlot")),
+
+        // Name of output EDM file if any.
+        // N.B. This parameter does not appear inside TMTrackProducer_Defaults_cfi.py . It is created inside
+        // tmtt_tf_analysis_cfg.py .
+        writeOutEdmFile_(iConfig.getUntrackedParameter<bool>("WriteOutEdmFile", true)),
+
+        // Bfield in Tesla. (Unknown at job initiation. Set to true value for each event
+        bField_(0.),
+
+        // Hybrid tracking
+        hybrid_(iConfig.getParameter<bool>("Hybrid")),
+        psStripPitch_(0.),
+        psNStrips_(0.),
+        psPixelLength_(0.),
+        ssStripPitch_(0.),
+        ssNStrips_(0.),
+        ssStripLength_(0.) {
+    // If user didn't specify any PDG codes, use e,mu,pi,K,p, to avoid picking up unstable particles like Xi-.
+    vector<unsigned int> genPdgIdsUnsigned(genCuts_.getParameter<vector<unsigned int>>("GenPdgIds"));
+    if (genPdgIdsUnsigned.empty()) {
+      genPdgIdsUnsigned = {11, 13, 211, 321, 2212};
+    }
+
+    // For simplicity, user need not distinguish particles from antiparticles in configuration file.
+    // But here we must store both explicitely in Settings, since TrackingParticleSelector expects them.
+    for (unsigned int i = 0; i < genPdgIdsUnsigned.size(); i++) {
+      genPdgIds_.push_back(genPdgIdsUnsigned[i]);
+      genPdgIds_.push_back(-genPdgIdsUnsigned[i]);
+    }
+
+    // Clean up list of fitters that require the r-z track filter to be run before them,
+    // by removing those fitters that are not to be run.
+    vector<string> useRZfilterTmp;
+    for (const string& name : useRZfilter_) {
+      if (std::count(trackFitters_.begin(), trackFitters_.end(), name) > 0)
+        useRZfilterTmp.push_back(name);
+    }
+    useRZfilter_ = useRZfilterTmp;
+
+    //--- Sanity checks
+
+    if (!(useStubPhi_ || useStubPhiTrk_))
+      throw cms::Exception(
+          "Settings.cc: Invalid cfg parameters - You cant set both UseStubPhi & useStubPhiTrk to false.");
```

Comment:
Exception categories here and below.

Reply:
Using category "BadConfig" (widely used in CMSSW)